### ISSUE TITLE
Provide shims as individual files

### ISSUE THREAD

**Issue by joshuacc** [es-shims/es5-shim#175]
It would be awesome if the shims were provided as individual files so that I could easily use only the shims that are necessary for my particular project.

**Comment by tomchentw** [comment]
I have tried to do a modular port using CommonJS and `webpack` as bundler. Here's the file for `Array.prototype.every`:
https://github.com/tomchentw-deprecated/holyfills/blob/master/lib/array/array_every.js

I'm interested to help on this. Any thoughts?

**Comment by ljharb** [comment]
Most every shim already exists as a standalone, and standalones will need to follow a certain convention I've yet to lay out. The difficulty is in constructing a build process.

While I appreciate the offer of help, I'm already working on this. I'll update here as I make progress.

**Comment by polkovnikov-ph** [comment]
Personally I think that it should be solved in a Common.js way: we should use only the things that are `require`d. A good example would be a module called `es5-shim/array` that could be used like

     var Array = require(`es5-shim/array`);

and it should contain all the internal stuff for `browserify` to translate it into either a piece of JS with all the browser support or plain `module.exports = Array` if there's no additional support needed.

Changing `prototype`s or defining global variables has never been a good idea. That's not the way a shim should work at all.

**Comment by hazzik** [comment]
>Changing prototypes or defining global variables has never been a good idea. That's not the way a shim should work at all.

It should work exactly this way. Shim provides polyfill for standard APIs which are not supported in some browsers

**Comment by oncletom** [comment]
Hello there :-)

Is it somethign which is still under consideration?

**Comment by ljharb** [comment]
@oncletom yes, it is. What I'm going to emulate is the process I've built in the es7-shim, but https://github.com/es-shims/es7-shim/issues/8 is a blocker before attempting that here. Please feel free to help on that issue on the es7-shim!
